src/language-js/printer-estree.js

```diff
@@ -1807,8 +1807,24 @@ function printPathNoParens(path, options, print, args) {
       if (n.value) {
         let res;
         if (isStringLiteral(n.value)) {
-          const value = rawText(n.value);
-          res = '"' + value.slice(1, -1).replace(/"/g, "&quot;") + '"';
+          const innerValue = rawText(n.value).slice(1, -1);
+          const hasSingle = innerValue.includes("'");
+          const hasDouble = innerValue.includes('"');
+
+          // If we need to escape quotes, just use the JS handler
+          if (hasSingle && hasDouble) {
+            res = path.call(print, "value");
```

Comment:
1. Does this mean that we would end up in the `printString` function in `src/common/util.js`? That would not be OK, because StringLiterals used as JSX attribute values do not have the same escaping rules as StringLiterals in other places.

2. Can this ever happen? Is it better to `throw` here?

Reply:
2. I'm not sure about this. I was under the impression that you can't escape quotes inside JSX string literal attributes, but the previous code handled escaping and there is also a comment above suggesting it might be possible. 

1. If escaping is not possible, then this can just be removed and not even worry about escaping. If it is, do you have some guidance on this? My first thought would be to just convert it into an actual expression and then use the underlying `printString` (from the above discussion we landed on using printString if an escape is needed to keep behavior parity)

Reply:
HTML escapes are used instead of backslashes.

Reply:
I update the above to make it a JSXExpressionContainer if an escape is needed so the `printString` function will properly escape now